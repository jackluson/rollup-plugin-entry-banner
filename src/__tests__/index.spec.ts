import bannerEntry, { createBanner, BannerEntryOptions } from '../index';
import { rollup } from 'rollup';

describe('rollup-plugin-entry-banner', () => {
  it('should match snapshot', async () => {
    const bundleWithPlugin = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
      plugins: [bannerEntry()],
    });
    const { output: outputWithPlugin } = await bundleWithPlugin.generate({ format: 'cjs' });
    expect(outputWithPlugin[0].code).toMatchSnapshot();
  });

  it('default config', async () => {
    const bundleWithPlugin = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
      plugins: [bannerEntry()],
    });
    const bundle = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
    });
    const { output: outputWithPlugin } = await bundleWithPlugin.generate({ format: 'cjs' });
    const { output } = await bundle.generate({ format: 'cjs' });
    expect(outputWithPlugin[0].code).toBe(createBanner() + output[0].code);
  });

  it('set preset config', async () => {
    const options = {
      preset: false,
    };
    const bundleWithPlugin = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
      plugins: [bannerEntry(options)],
    });
    const bundle = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
    });
    const { output: outputWithPlugin } = await bundleWithPlugin.generate({ format: 'cjs' });
    const { output } = await bundle.generate({ format: 'cjs' });
    expect(outputWithPlugin[0].code).toBe(createBanner(options) + output[0].code);
  });

  it('set renderBanner config', async () => {
    const options1 = {
      renderBanner: (code: string) => code + '/**/',
    };
    async function testRenderBanner(options: BannerEntryOptions) {
      const bundleWithPlugin = await rollup({
        input: 'src/__tests__/fixtures/index.js',
        output: {
          exports: 'default',
        },
        plugins: [bannerEntry(options)],
      });
      const bundle = await rollup({
        input: 'src/__tests__/fixtures/index.js',
        output: {
          exports: 'default',
        },
      });
      const { output: outputWithPlugin } = await bundleWithPlugin.generate({ format: 'cjs' });
      const { output } = await bundle.generate({ format: 'cjs' });
      expect(outputWithPlugin[0].code).toBe(createBanner(options) + output[0].code);
    }
    testRenderBanner(options1);
  });

  it('set packageJsonDir config', async () => {
    const options = {
      packageJsonDir: process.cwd() + '/src/__tests__/fixtures',
    };
    const bundleWithPlugin = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
      plugins: [bannerEntry(options)],
    });
    const bundle = await rollup({
      input: 'src/__tests__/fixtures/index.js',
      output: {
        exports: 'default',
      },
    });
    const { output: outputWithPlugin } = await bundleWithPlugin.generate({ format: 'cjs' });
    const { output } = await bundle.generate({ format: 'cjs' });
    expect(outputWithPlugin[0].code).toBe(createBanner(options) + output[0].code);
  });
});
