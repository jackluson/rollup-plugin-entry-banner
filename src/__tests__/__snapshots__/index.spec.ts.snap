// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-entry-banner should match snapshot 1`] = `
";console.info("%c Name: %c rollup-plugin-entry-banner ", "background:#606060; padding: 1px; border-radius: 3px 0 0 3px; color: #fff", "background:#42c02e; padding: #1px; border-radius: 0 3px 3px 0; color: #fff");console.info("%c Version: %c 1.0.1 ", "background:#35495e; padding: 1px; border-radius: 3px 0 0 3px; color: #fff", "background:#41b883; padding: #1px; border-radius: 0 3px 3px 0; color: #fff");console.info("%c Node Version: %c v16.15.1 ", "background:#35495e; padding: 1px; border-radius: 3px 0 0 3px; color: #fff", "background:#41b883; padding: #1px; border-radius: 0 3px 3px 0; color: #fff");console.info("%c Built at: %c 2022/12/15 19:18:06 ", "background:#35495e; padding: 1px; border-radius: 3px 0 0 3px; color: #fff", "background:#41b883; padding: #1px; border-radius: 0 3px 3px 0; color: #fff");'use strict';

function testEntryBanner(){}

module.exports = testEntryBanner;
"
`;
